packages/grafana-ui/src/types/datasource.ts

```diff
@@ -89,12 +89,7 @@ export class DataSourcePlugin<
   setComponentsFromLegacyExports(pluginExports: any) {
     this.angularConfigCtrl = pluginExports.ConfigCtrl;
 
-    this.components.QueryCtrl = pluginExports.QueryCtrl;
-    this.components.AnnotationsQueryCtrl = pluginExports.AnnotationsQueryCtrl;
-    this.components.ExploreQueryField = pluginExports.ExploreQueryField;
-    this.components.ExploreStartPage = pluginExports.ExploreStartPage;
-    this.components.QueryEditor = pluginExports.QueryEditor;
-    this.components.VariableQueryEditor = pluginExports.VariableQueryEditor;
+    this.components = omit(pluginExports, ['ConfigCtrl', 'Datasource']);
```

Comment:
The support for ConfigEditor react component is already there but because of how things were mapped here it would not be shown in components.

Reply:
Not sure I understand, ConfigEditor should not be exposed via export but via DataSourcePlugin export (see testdata and input data source plugins for examples)

Reply:
Is there a reason this should __not__ work if you do not use the `DataSourcePlugin`? I mean it does the same thing:
```
    this.components.ConfigEditor = editor;
```
so the important part is only the naming not whether you use the `DataSourcePlugin` or the standard exports.

Reply:
yes. The reason is that this is a new component and not part of the old legacy set of components. We want data sources to export just DataSourcePlugin, not a just a bunch of plain exports that we have no way to typecheck.

Reply:
Ok will migrate loki to the `DataSourcePlugin`